### ISSUE TITLE
PDF inconsistent  in generation

### ISSUE THREAD

**Issue by dawez** [rkusa/pdfjs#124]
Following this ticket #118 it seems that I am also unable to have the PDF consistent on each generation. For example I am trying to add 3 images to a pdf and have a single image per page.

The code I have so far is:

`const fetch = require ("node-fetch");

const pdf = require("pdfjs");
const { forEach } = require("p-iteration");
const fs = require("fs");

const createPDF = async () => {
  const images = [
    "https://raw.githubusercontent.com/drewnoakes/metadata-extractor-images/master/jpg/Canon%20PowerShot%20A610.jpg",
    "https://raw.githubusercontent.com/drewnoakes/metadata-extractor-images/master/jpg/Apple%20iPhone%206%20plus.jpg",
    "https://raw.githubusercontent.com/drewnoakes/metadata-extractor-images/master/jpg/Android%20Depth%20Map.jpg"
  ];

  const doc = new pdf.Document();
  const lastImageUrl = images[images.length - 1];

  const w = fs.createWriteStream("/tmp/out.pdf");
  doc.pipe(w);

  await forEach(images, async imageUrl => {
    const response = await fetch(imageUrl);
    console.log(response.size);
    const buffer = await response.buffer();
    await doc.image(new pdf.Image(buffer));
    
    if (imageUrl != lastImageUrl) {
      doc.pageBreak();
    }
  });

  const done = new Promise(resolve => {
    console.log("finish");
    w.on("finish", resolve);
  });
  await doc.end();
  await done;
};

createPDF().then(value => {
  console.log("done");
});
`

Basically it pulls 3 images and add one image per page. It happens that sometimes I have all the 3 images in the output and sometimes only 3 [ the last page is empty]

I tried the suggestion posted in #118 but still have cases of 2 images instead of 3. Is there anything that I can check here?

**Comment by dawez** [comment]
It seems that the problem was due the iteration I used. Changing with:

`const images = [
    "https://raw.githubusercontent.com/drewnoakes/metadata-extractor-images/master/jpg/Apple%20iPhone%206%20plus.jpg",
    "https://raw.githubusercontent.com/drewnoakes/metadata-extractor-images/master/jpg/Android%20Depth%20Map.jpg",
    "https://raw.githubusercontent.com/drewnoakes/metadata-extractor-images/master/jpg/Canon%20PowerShot%20A610.jpg"
  ];

  const doc = new pdf.Document();
  const w = fs.createWriteStream("/tmp/out.pdf");
  doc.pipe(w);

  for (let i = 0; i < images.length; i++) {
    const imageUrl = images[i];
    const response = await fetch(imageUrl);
    const buffer = await response.buffer();
    doc.image(new pdf.Image(buffer));

    if (i < images.length - 1) {
      doc.pageBreak();
    }
  }

  const done = new Promise(resolve => w.on("finish", resolve));
  await doc.end();
  await done;`

seems to be working all the times.. go figure...
